server/rest/src/main/java/org/infinispan/rest/resources/CounterResource.java

```diff
@@ -43,32 +44,23 @@ public Invocations getInvocations() {
             .create();
    }
 
-   private NettyRestResponse getCounter(RestRequest request) throws RestResponseException {
-      try {
-         String counterName = request.variables().get("counterName");
-         String accept = request.getAcceptHeader();
-         MediaType contentType = accept == null ? MediaType.TEXT_PLAIN : negotiateMediaType(accept);
-         NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
-         CounterConfiguration configuration = counterManager.getConfiguration(counterName);
-         if (configuration == null) {
-            return responseBuilder.status(HttpResponseStatus.NOT_FOUND.code()).entity("Counter not found").build();
+   private CompletableFuture<RestResponse> getCounter(RestRequest request) throws RestResponseException {
+      String counterName = request.variables().get("counterName");
+      String accept = request.getAcceptHeader();
+      MediaType contentType = accept == null ? MediaType.TEXT_PLAIN : negotiateMediaType(accept);
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      CounterConfiguration configuration = counterManager.getConfiguration(counterName);
+      if (configuration == null) {
+         return CompletableFuture.completedFuture(responseBuilder.status(HttpResponseStatus.NOT_FOUND.code()).entity("Counter not found").build());
+      } else {
+         responseBuilder.contentType(contentType).header(HttpHeaderNames.CACHE_CONTROL.toString(), CacheControl.noCache());
+         CompletableFuture<Long> response;
+         if (configuration.type() == CounterType.WEAK) {
+            response = CompletableFuture.completedFuture(counterManager.getWeakCounter(counterName).getValue());
```

Comment:
No need to wrap the value, it is already a `CompletableFuture<Long>`.

Reply:
Not for the [WeakCounter](https://github.com/infinispan/infinispan/blob/759e8359d315b7d5eb80f334443b1ab3990640d6/commons/src/main/java/org/infinispan/counter/api/WeakCounter.java#L30)